examples/helpers/tiled.rs

```diff
@@ -135,115 +181,159 @@ pub fn process_loaded_maps(
-                                    continue;
+                                        let texture_index = match tileset.image {
+                                            Some(_) => TileTextureIndex(layer_tile_data.id()),
+                                            None => tiled_map
+                                                .tile_texture_indexes
+                                                .get(&(tileset_index, layer_tile.id()))
+                                                .expect(
+                                                    "The tile should have been mapped previously.",
+                                                )
+                                                .clone(),
+                                        };
+
+                                        let tile_pos = TilePos { x, y };
+                                        let tile_entity = commands
+                                            .spawn(TileBundle {
+                                                position: tile_pos,
+                                                tilemap_id: TilemapId(layer_entity),
+                                                texture_index: texture_index,
+                                                flip: TileFlip {
+                                                    x: layer_tile_data.flip_h,
+                                                    y: layer_tile_data.flip_v,
+                                                    d: layer_tile_data.flip_d,
+                                                },
+                                                ..Default::default()
+                                            })
+                                            .id();
+                                        tile_storage.set(&tile_pos, tile_entity);
+                                    }
                                 }
 
-                                let tile_id = map_tile.gid - tileset.first_gid;
-
-                                let tile_pos = TilePos { x, y };
-                                let tile_entity = commands
-                                    .spawn(TileBundle {
-                                        position: tile_pos,
-                                        tilemap_id: TilemapId(layer_entity),
-                                        texture_index: TileTextureIndex(tile_id),
-                                        flip: TileFlip {
-                                            x: map_tile.flip_h,
-                                            y: map_tile.flip_v,
-                                            d: map_tile.flip_d,
-                                        },
-                                        ..Default::default()
-                                    })
-                                    .id();
-                                tile_storage.set(&tile_pos, tile_entity);
+                                let texture = if tiled_map.tile_texture_indexes.is_empty() {
```

Comment:
It seems like it should be possible to allow "image collection tilesets" as well as normal tilesets in a single map, provided each layer exclusively uses one or the other type of tileset.

But this code causes `TilemapTexture::Vector` to be used for every layer if any layer uses an image collection tileset.